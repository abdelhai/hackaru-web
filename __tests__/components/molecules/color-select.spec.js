import Factory from '@/__tests__/__setups__/factory';
import ColorSelect from '@/components/molecules/color-select';

describe('ColorSelect', () => {
  let factory;
  let wrapper;

  beforeEach(() => {
    factory = new Factory(ColorSelect);
  });

  it('render correctly', () => {
    expect(factory.shallow().element).toMatchSnapshot();
  });

  describe('when color select', () => {
    beforeEach(() => {
      wrapper = factory.shallow();
      wrapper.setProps({ colors: ['#ff0', '#f00', '#0ff'] });
      wrapper
        .findAll('button')
        .at(0)
        .trigger('click');
    });

    it('emit update:value', () => {
      expect(wrapper.emitted('update:value')[0]).toEqual(['#ff0']);
    });
  });
});
