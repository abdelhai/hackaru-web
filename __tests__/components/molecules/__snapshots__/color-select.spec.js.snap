// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ColorSelect render correctly 1`] = `
<section>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#4ab8b8"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#7da87b"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#a1c45a"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#7a4579"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#f95959"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#ea8a8a"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#ff9f68"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#facf5a"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#616161"
    />
  </button>
  <button
    class=""
    type="button"
  >
    <dot-stub
      class="is-large"
      color="#8ea6b4"
    />
  </button>
</section>
`;
